src/main/java/io/nats/client/Connection.java

```diff
@@ -193,6 +193,20 @@
      */
     CompletableFuture<Message> request(Message message);
 
+    /**
+     * Send a request. The returned future will be completed when the
+     * response comes back.
+     *
+     * <p>The Message object allows you to set a replyTo, but in requests,
+     * the replyTo is reserved for internal use as the address for the
+     * server to respond to the client with the consumer's reply.</p>
+     *
+     * @param message the message
+     * @param timeout the time to wait for a response
+     * @return a Future for the response, which may be cancelled on error or timed out
+     */
+    CompletableFuture<Message> requestWithTimeout(Message message, Duration timeout);

```

Comment:
For parity, I think we'd also want:
```java
CompletableFuture<Message> requestWithTimeout(String subject, byte[] body, Duration timeout);
```

wdyt @scottf ?

Reply:
@dxxyy please add.

Reply:
will do!